lib/datasets/dataset.rb

```diff
@@ -33,9 +33,21 @@ def cache_path
       @cache_path ||= CachePath.new(@metadata.id)
     end
 
-    def download(output_path, url, &block)
-      downloader = Downloader.new(url)
-      downloader.download(output_path, &block)
+    def download(output_path, *urls, &block)
```

Comment:
How about requiring at least one URL?

```suggestion
    def download(output_path, url, *fallback_urls, &block)
      urls = [url] + fallback_urls
```

Reply:
I agree, at least one URL should be required.

Reply:
I implemented in aafbddb.